pkg/autodiscovery/providers/kubecontainer.go

```diff
@@ -58,212 +49,158 @@ func (k *KubeContainerConfigProvider) String() string {
 	return names.KubeContainer
 }
 
-// Collect retrieves all running pods and extract AD templates from their annotations.
+// Collect is not implemented
 func (k *KubeContainerConfigProvider) Collect(ctx context.Context) ([]integration.Config, error) {
-	k.once.Do(func() {
-		go k.listen()
-	})
-
-	k.Lock()
-	k.upToDate = true
-	k.Unlock()
-
-	return k.generateConfigs()
+	return nil, nil
 }
 
-func (k *KubeContainerConfigProvider) listen() {
+// Stream starts listening to workloadmeta to generate configs as they come
+// instead of relying on a periodic call to Collect.
+func (k *KubeContainerConfigProvider) Stream(ctx context.Context) <-chan integration.ConfigChanges {
 	const name = "ad-kubecontainerprovider"
 
-	k.Lock()
-	k.streaming = true
-	health := health.RegisterLiveness(name)
-	defer func() {
-		err := health.Deregister()
-		if err != nil {
-			log.Warnf("error de-registering health check: %s", err)
-		}
-	}()
-	k.Unlock()
+	outCh := make(chan integration.ConfigChanges)
```

Comment:
I think it's critical that this is unbuffered, in order to ensure no races with the listener, right?  Worth a comment?